coordinator/src/node/channel.rs

```diff
@@ -285,4 +338,179 @@ impl Node {
+                )? > 0
+                {
+                    tracing::info!(%trader_id, "Set closing position to closed after the dlc channel got force closed.");
+                } else {
+                    tracing::warn!(%trader_id, "Failed to set closing position to closed after the dlc channel got force closed.");
+                }
+            }
+            DlcChannelEvent::CollaborativelyClosed(_) => {
+                let channel = &self.inner.get_dlc_channel_by_reference_id(reference_id)?;
+                let protocol_executor = dlc_protocol::DlcProtocolExecutor::new(self.pool.clone());
+                protocol_executor.finish_dlc_protocol(
+                    protocol_id,
+                    &to_secp_pk_30(channel.get_counter_party_id()),
+                    None,
+                    &channel.get_id(),
+                    self.tx_position_feed.clone(),
+                )?;
+            }
+            DlcChannelEvent::Offered(_)
+            | DlcChannelEvent::Accepted(_)
+            | DlcChannelEvent::Established(_)
+            | DlcChannelEvent::SettledOffered(_)
+            | DlcChannelEvent::SettledReceived(_)
+            | DlcChannelEvent::SettledAccepted(_)
+            | DlcChannelEvent::SettledConfirmed(_)
+            | DlcChannelEvent::Settled(_)
+            | DlcChannelEvent::SettledClosing(_)
+            | DlcChannelEvent::RenewOffered(_)
+            | DlcChannelEvent::RenewAccepted(_)
+            | DlcChannelEvent::RenewConfirmed(_)
+            | DlcChannelEvent::RenewFinalized(_)
+            | DlcChannelEvent::CollaborativeCloseOffered(_)
+            | DlcChannelEvent::ClosedPunished(_)
+            | DlcChannelEvent::FailedAccept(_)
+            | DlcChannelEvent::FailedSign(_)
+            | DlcChannelEvent::Cancelled(_)
+            | DlcChannelEvent::Deleted(_) => {} // ignored
+        }
+
+        Ok(())
+    }
+
+    /// Calculates the trader realized pnl from the cet outputs which do not belong to us.
+    /// 1. Sum the trader payouts
+    /// 2. Subtract the trader reserve sats from the trader payout
+    fn calculate_trader_realized_pnl_from_cet(
+        &self,
+        conn: &mut PooledConnection<ConnectionManager<PgConnection>>,
+        channel_id: &DlcChannelId,
+        signed_cet: Transaction,
+    ) -> Result<i64> {
+        let trader_payout: u64 = signed_cet
+            .output
+            .iter()
+            .filter(|output| {
+                !self
+                    .inner
+                    .is_mine(&ScriptBuf::from_bytes(output.script_pubkey.to_bytes()))
+            })
+            .map(|output| output.value)
+            .sum();
+
+        let dlc_channel =
+            db::dlc_channels::get_dlc_channel(conn, channel_id)?.with_context(|| {
+                format!("Couldn't find dlc channel by channel id = {:?}", channel_id)
+            })?;
+
+        let trader_realized_pnl_sat =
+            trader_payout as i64 - dlc_channel.trader_reserve_sats.to_sat() as i64;
```

Comment:
🤔 I don't think this is how we calculate PNL.

The payout is already `collateral_reserve + trader_margin +/- pnl`. So to get the PNL you would have to subtract `collateral_reserve` and `trader_margin`.

Reply:
Nice finding, I'll create a separate ticket for it since this is already in main.

Reply:
Haha, I thought it was new code :O

Reply:
https://github.com/get10101/10101/issues/2524